statsmodels/discrete/discrete_margins.py

```diff
@@ -446,18 +446,43 @@ def summary_frame(self, alpha=.05):
         -------
         frame : DataFrames
             A DataFrame summarizing the marginal effects.
+
+        Notes
+        -----
+        The dataframe is created on each call and not cached, as are the
+        tables build in `summary()`
         """
         _check_at_is_all(self.margeff_options)
-        from pandas import DataFrame
+        results = self.results
+        model = self.results.model
+        from pandas import DataFrame, MultiIndex
         names = [_transform_names[self.margeff_options['method']],
                                   'Std. Err.', 'z', 'Pr(>|z|)',
                                   'Conf. Int. Low', 'Cont. Int. Hi.']
         ind = self.results.model.exog.var(0) != 0 # True if not a constant
         exog_names = self.results.model.exog_names
         var_names = [name for i,name in enumerate(exog_names) if ind[i]]
-        table = np.column_stack((self.margeff, self.margeff_se, self.tvalues,
-                                 self.pvalues, self.conf_int(alpha)))
-        return DataFrame(table, columns=names, index=var_names)
+
+        if self.margeff.ndim == 2:
+            # MNLogit case
+            table = np.column_stack([i.ravel("F") for i in
+                        [self.margeff, self.margeff_se, self.tvalues,
+                         self.pvalues, self.conf_int(alpha)[:, 0, :],
+                         self.conf_int(alpha)[:, 1, :]]])
+
+            _, yname_list = results._get_endog_name(model.endog_names,
+                                                        None, all=True)
+            ynames = np.repeat(yname_list, len(var_names))
+            xnames = np.tile(var_names, len(yname_list))
+            index = MultiIndex.from_tuples(list(zip(ynames, xnames)),
+                                           names=['endog', 'exog'])
+        else:
+            table = np.column_stack((self.margeff, self.margeff_se, self.tvalues,
+                                     self.pvalues, self.conf_int(alpha)[:, 0, :]))
```

Comment:
BUG: the slicing here is left over from experimenting.
no error raise in unit tests!